cgo/helpers.go

```diff
@@ -0,0 +1,287 @@
+package cgo
+
+/*
+#cgo LDFLAGS: -L${SRCDIR}/..
+#cgo pkg-config: ${SRCDIR}/../filcrypto.pc
+#include "../filcrypto.h"
+#include <stdlib.h>
+*/
+import "C"
+import (
+	"errors"
+	"unsafe"
+)
+
+func AsSliceRefUint8(goBytes []byte) SliceRefUint8 {
+	len := len(goBytes)
+
+	if len == 0 {
+		// can't take element 0 of an empty slice
+		return SliceRefUint8{
+			ptr: (*C.uint8_t)(unsafe.Pointer(&goBytes)),
```

Comment:
I would feel better if we had a global `uint8` variable and were taking pointer to it instead of the slice.
Reinterpreting the slice pointer to `C.uint8_t` feels wrong.

Reply:
But it would make the code more verbose (all different types).
Alternative would be the pointer to be `nil`.

Reply:
I don't understand the concern/solution I am afraid.

Reply:
@Kubuxu can you re-explain this if it's still a concern?

Reply:
explained in person, fix just pushed